.. _presentation-materials:

=====================
COPO Presentations
=====================

This section entails presentations and other materials that have been used to introduce COPO to various audiences.

.. hint::

    * Click the download link below to access the presentations

    * View the seminar presentations in slide show mode to see the animations.

.. raw:: html

   <hr>

Seminar presentations
----------------------

#. **New COPO Tools for Brokering Single-cell & Spatial Omics Metadata**

   - Presented at the Earlham Institute internal seminar

   - **Date**: Tuesday 1st July, 2025

   - **Presenters**: Debby Ku & Aaliyah Providence, Research Software Engineers of the COPO team

   - **Abstract**:
      The Collaborative OPen Omics (**COPO**) platform now includes dedicated support for submitting metadata from
      Single-cell and Spatial Transcriptomics experiments. Whether you're working with Single-cell Ribonucleic Acid
      Sequencing  (**scRNA-Seq**), Spatial Transcriptomics Fluorescence In Situ Hybridisation (**ST-FISH**) or Spatial
      Transcriptomics Sequencing (**ST-Seq**), COPO streamlines the process of structuring, validating, and submitting
      your metadata using established standards such as Darwin Core (**DwC**), Minimum Information about any (x)
      Sequence (**MIxS**) and Functional Annotation of Animal Genomes (**FAANG**).

      This new configuration, tailored for single-cell workflows, also serves as a flexible tool that can be adapted
      to meet a wide range of metadata requirements across diverse omics domains. COPO does not enforce a fixed
      metadata model - instead, it supports user-defined metadata structures, making it easy to extend or modify
      worksheets, add fields and build manifests that suit your specific data submission types.

      With integration to public repositories like the European Nucleotide Archive (ENA) and Zenodo, COPO helps ensure
      that your data is Findable, Accessible, Interoperable and Reusable (FAIR) with the scientific community.

      Join this seminar to see how COPO can save you time, reduce submission errors and support your single-cell
      metadata submissions - while remaining fully adaptable for any other metadata needs you may have.


   - :download:`Download seminar presentation <../assets/files/presentations/EI_Seminar_01072025_New_COPO_Tools_for_Brokering_Single_cell_and _Spatial_Omics_Metadata.pptx>`

   .. raw:: html

         <hr>

#. **Leveraging COPO to submit READ/Assembly/Sequencing Annotation data to ENA and get data published**

   - Presented at the Earlham Institute internal seminar

   - **Date**: Tuesday 14th May, 2024

   - **Presenter**: Debby Ku, Research Software Engineer of the COPO team

   - **Abstract**:
      How users can leverage COPO to submit reads, assemblies and sequencing annotation data to European Nucleotide
      Archive (ENA) and get the data published

   - :download:`Download seminar presentation <../assets/files/presentations/EI_Seminar_14052024_Leveraging_COPO_to_submit_READ_Assembly_Sequencing_Annotation_data_to_ENA_and_get_data_published.pptx>`

   .. raw:: html

         <hr>

#. **Advancing Biodiversity Research: The Evolution of Collaborative OPen Omics (COPO) Project**

   - Presented at the Earlham Institute internal seminar

   - **Date**: Tuesday 23rd April, 2024

   - **Presenter**: Miss Aaliyah Providence, Junior Software Developer of the COPO team

   - **Abstract**:
      This seminar explores the evolution of the Collaborative OPen Omics (COPO) project as an
      open-source metadata broker in biodiversity research. With its capability to broker
      different data seamlessly, COPO has revolutionised the way by which researchers collaborate
      and share information.

      An integral part of COPO's development involves standardising as well as enriching metadata
      and enabling the integration of diverse datasets for publication to the European Nucleotide
      Archive (ENA) which facilitates public dissemination.

      Additionally, COPO's user-friendly Application Programming Interface (API) empowers users
      to query sample data efficiently, further enhancing accessibility and interoperability in
      biodiversity studies. Through its continuous evolution, COPO continues to drive innovation
      and collaboration in the field of bioscience research.

   - :download:`Download seminar presentation <../assets/files/presentations/EI_Seminar_23042024_Advancing_Biodiversity_Research_The_Evolution_of_COPO.pptx>`

.. raw:: html

   <hr>

Poster presentations
----------------------

#. **Collaborative OPen Omics (COPO) - Facilitating Metadata Interoperability with Standards, Formats and APIs**

   - **Authors**: COPO team - Aaliyah Providence, Debby Ku and Felix Shaw

   - **Abstract**:
      The Collaborative OPen Omics (COPO) platform is a data and metadata broker for researchers seeking to describe
      and share a wide range of research objects - including genomic reads, assemblies, sequence annotations, samples
      and images—using standardised metadata. By enabling users to submit metadata through well-defined spreadsheets
      and web forms, COPO facilitates the publication of this data to public repositories such as the European
      Nucleotide Archive (ENA) and the National Centre for Biotechnology Information (NCBI). COPO’s integration of
      community-derived schemas and vocabularies ensures that data is Findable, Accessible, Interoperable, and Reusable
      (FAIR).

      COPO can validate and convert submitted metadata into various standards, making it accessible through its
      Application Programming Interface (API). This process allows for seamless mapping of metadata from one standard,
      such as Tree of Life (ToL), to other recognised standards like Darwin Core (DwC) and Minimum Information about
      any (x) Sequence (MIxS). Researchers can retrieve this metadata in structured formats such as Research Object
      Crate (RO-Crate), JavaScript Object Notation (JSON) or Comma-Separated Values (CSV), enhancing data
      discoverability and reuse.

      Our poster illustrates how COPO supports interoperability across different systems by brokering and sharing
      metadata, while also ensuring the exclusion of sensitive data such as ORCiD identifiers and email addresses.
      By empowering researchers to securely submit and manage their data in compliance with recognised standards,
      COPO fosters open science and advances research in the life sciences.

   - :download:`Download poster presentation <../assets/files/presentations/copo_poster_enhancing_metadata_interoperability_with_standards_formats_and_apis.pdf>`

   .. raw:: html

         <hr>

#. **Collaborative OPen Omics (COPO) - Building, Testing and Deployment**

   - **Authors**: COPO team - Aaliyah Providence, Debby Ku and Felix Shaw

   - **Abstract**:
      Collaborative OPen Omics (COPO) is a user-friendly platform that allows scientists to easily upload, update, and
      publish metadata while adhering to community standards, promoting transparent data sharing. Built with
      cutting-edge technology, it utilises Docker Swarm for seamless service management, ensuring continuous
      availability, scalability, and ease of maintenance.

      Security is prioritised in COPO's architecture, with distinct network layers preventing direct client access
      to back-end data repositories. Docker secrets are employed to enforce strict measures against unauthorised
      access, enhancing data protection and platform integrity.

      Overall, COPO provides a sophisticated solution for scientific data management, integrating modern methodologies
      and security protocols to facilitate collaborative research efforts and promote scientific advancement.

   - :download:`Download poster presentation <../assets/files/presentations/copo_poster_building_testing_and_deployment.pdf>`

   .. raw:: html

         <hr>

#. **COPO - Bridging the Gap from Data to Publication in Plant Science**

   - **Authors**: Felix Shaw, Anthony Etuk, Alejandra Gonzalez-Beltran, Phillipe Rocca-Serra, Alfie Abdul-Rahman,
     Paul Kersey, Ruth Bastow, Susanna Sansone, Robert Davey

   - **Abstract**:
      COPO streamlines the process of data deposition to public repositories by hiding much of the complexity of
      metadata capture and data management from the end-user. The ISA infrastructure (www.isa-tools.org) is leveraged
      to provide the interoperability between metadata formats required for seamless deposition to repositories and to
      facilitate links to data analysis platforms.

      Logical groupings of artefacts (e.g. PDFs, raw data, contextual
      supplementary information) relating to a body of work are stored in COPO collections and represented by common
      standards, which are publicly searchable. Bundles of multiple data objects themselves can then be deposited
      directly into public repositories through COPO interfaces.



   - :download:`Download poster presentation <https://figshare.com/ndownloader/files/2175949>`

.. raw:: html

   <hr>

.. seealso::

   * :ref:`COPO FAIR webinar presentations <presentations-webinars-fair>`
   * :download:`Download general Tree of Life visual submission documentation <../assets/files/COPO_visual_user_documentation.pdf>`
   * :download:`Download an illustration of samples submission and validation process in COPO <../assets/files/presentations/copo_sample_submission_process_illustration.gif>`
